stellar_sdk/soroban_rpc.py

```diff
@@ -75,16 +75,27 @@ class PaginationOptions(BaseModel):
     limit: Optional[int] = None
 
 
-class GetEventsRequest(BaseModel):
+class PaginationMixin:
+    pagination: Optional[PaginationOptions] = None
+
+    @model_validator(mode="after")
+    def verify_ledger_or_cursor(self) -> Self:
```

Comment:
```suggestion
    def verify_ledger_or_cursor(self) -> "PaginationMixin":
```

We need to support lower versions of Python.

Reply:
Would it be ok to use `typing_extensions` instead? It is guaranteed to be available because Pydantic depends on it.
The advantage would be that it's clearer and that it will be more obvious what needs to happen as you drop support for older Python versions.

Reply:
👍 good idea.

Reply:
Would it be better to add `typing_extensions` to the dependencies? Because we cannot guarantee that `pydantic` will always depend on it.

Reply:
ok, i'll do that